core/src/main/java/ai/timefold/solver/core/impl/score/stream/bavet/common/AbstractJoinNode.java

```diff
@@ -61,28 +73,6 @@ protected final void insertOutTuple(LeftTuple_ leftTuple, UniTuple<Right_> right
         propagationQueue.insert(outTuple);
     }
 
-    protected final void insertOutTupleFiltered(LeftTuple_ leftTuple, UniTuple<Right_> rightTuple) {
-        if (!isFiltering || testFiltering(leftTuple, rightTuple)) {
-            insertOutTuple(leftTuple, rightTuple);
-        }
-    }
-
-    protected final void innerUpdateLeft(LeftTuple_ leftTuple, Consumer<Consumer<UniTuple<Right_>>> rightTupleConsumer) {
-        // Prefer an update over retract-insert if possible
-        ElementAwareList<OutTuple_> outTupleListLeft = leftTuple.getStore(inputStoreIndexLeftOutTupleList);
-        // Propagate the update for downstream filters, matchWeighers, ...
-        if (!isFiltering) {
-            for (var outTuple : outTupleListLeft) {
-                updateOutTupleLeft(outTuple, leftTuple);
-            }
-        } else {
-            rightTupleConsumer.accept(rightTuple -> {
```

Comment:
This lambda will no longer be created. In my benchmarks, this took gigabytes of heap and therefore also GC cycles.